pkg/controller/scheduler.go

```diff
@@ -834,7 +847,8 @@ func (c *Controller) checkCanaryStatus(canary *flaggerv1.Canary, canaryControlle
 
 func (c *Controller) hasCanaryRevisionChanged(canary *flaggerv1.Canary, canaryController canary.Controller) bool {
 	if canary.Status.Phase == flaggerv1.CanaryPhaseProgressing ||
-		canary.Status.Phase == flaggerv1.CanaryPhaseWaitingPromotion {
+		canary.Status.Phase == flaggerv1.CanaryPhaseWaitingPromotion ||
```

Comment:
I don't think we need to check for `CanaryPhaseFrozen`, because of the concern described in the above comment.